go/vt/vtgate/planbuilder/operators/horizon_expanding.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2023 The Vitess Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operators
+
+import (
+	"fmt"
+
+	"vitess.io/vitess/go/vt/sqlparser"
+	"vitess.io/vitess/go/vt/vterrors"
+	"vitess.io/vitess/go/vt/vtgate/planbuilder/operators/ops"
+	"vitess.io/vitess/go/vt/vtgate/planbuilder/operators/rewrite"
+	"vitess.io/vitess/go/vt/vtgate/planbuilder/plancontext"
+)
+
+func expandHorizon(ctx *plancontext.PlanningContext, horizon horizonLike) (ops.Operator, *rewrite.ApplyResult, error) {
+	sel, isSel := horizon.selectStatement().(*sqlparser.Select)
+	if !isSel {
+		return nil, nil, errHorizonNotPlanned()
+	}
+
+	if sel.Having != nil {
+		return nil, nil, errHorizonNotPlanned()
+	}
```

Comment:
```suggestion
	if !isSel || sel.Having != nil {
		return nil, nil, errHorizonNotPlanned()
	}
```

Reply:
having PR is already in the works that will remove this check